src/cli/src/consumer/produce/mod.rs

```diff
@@ -56,12 +56,19 @@ impl ProduceOpt {
 
         match &self.file {
             Some(path) => {
-                let mut reader = BufReader::new(File::open(path)?);
-                self.produce_lines(&mut producer, &mut reader).await?;
+                let reader = BufReader::new(File::open(path)?);
+                let lines: Vec<_> = reader.lines().filter_map(|it| it.ok()).collect();
+                let batches: Vec<_> = lines.chunks(50).collect();
```

Comment:
I think it is possible to combine `lines`,`batches` into a single iterator

Reply:
I believe that `chunks` is a method on slices, not on iterators. So I needed to collect into a Vec first to make this work